pkg/model/resource/resource.go

```diff
@@ -78,10 +137,94 @@ func (r Resource) Replacer() *strings.Replacer {
 	var replacements []string
 
 	replacements = append(replacements, wrapKey("group"), r.Group)
-	replacements = append(replacements, wrapKey("group-package-name"), r.GroupPackageName)
 	replacements = append(replacements, wrapKey("version"), r.Version)
 	replacements = append(replacements, wrapKey("kind"), strings.ToLower(r.Kind))
 	replacements = append(replacements, wrapKey("plural"), strings.ToLower(r.Plural))
+	replacements = append(replacements, wrapKey("package-name"), r.PackageName())
 
 	return strings.NewReplacer(replacements...)
 }
+
+// API holds the information related to the golang type definition and the CRD.
+type API struct {
+	// Version holds the CustomResourceDefinition API version used for the resource.
+	Version string `json:"crdVersion,omitempty"`
+
+	// Namespaced is true if the resource is namespaced.
+	Namespaced bool `json:"namespaced,omitempty"`
+}
+
+// update combines fields of the APIs of two resources.
+func (api *API) update(other *API) error {
+	// If other is nil, nothing to merge
+	if other == nil {
+		return nil
+	}
+
+	// Update the version.
+	if other.Version != "" {
+		if api.Version == "" {
+			api.Version = other.Version
+		} else if api.Version != other.Version {
+			return fmt.Errorf("CRD versions do not match")
+		}
+	}
+
+	// Update the namespace.
+	api.Namespaced = api.Namespaced || other.Namespaced
+
+	return nil
+}
+
+// IsEmpty returns if the API's fields all contain zero-values.
+func (api API) IsEmpty() bool {
+	return api.Version == "" && !api.Namespaced
+}
+
+// Webhooks holds the information related to the associated webhooks.
+type Webhooks struct {
+	// Version holds the {Validating,Mutating}WebhookConfiguration API version used for the resource.
+	Version string `json:"webhookVersion,omitempty"`
+
+	// Defaulting specifies if a defaulting webhook is associated to the resource.
+	Defaulting bool `json:"defaulting,omitempty"`
+
+	// Validation specifies if a validation webhook is associated to the resource.
+	Validation bool `json:"validating,omitempty"`
+
+	// Conversion specifies if a conversion webhook is associated to the resource.
+	Conversion bool `json:"conversion,omitempty"`
+}
+
+// update combines fields of the webhooks of two resources.
+func (webhooks *Webhooks) update(other *Webhooks) error {
+	// If other is nil, nothing to merge
+	if other == nil {
+		return nil
+	}
+
+	// Update the version.
+	if other.Version != "" {
+		if webhooks.Version == "" {
+			webhooks.Version = other.Version
+		} else if webhooks.Version != other.Version {
+			return fmt.Errorf("webhook versions do not match")
+		}
+	}
```

Comment:
```suggestion
		if webhooks.WebhookVersion == "" {
			webhooks.WebhookVersion = other.WebhookVersion
		} else if webhooks.WebhookVersion != other.WebhookVersion {
			return fmt.Errorf("webhook versions do not match")
		}
	}
```

Reply:
Same as in https://github.com/kubernetes-sigs/kubebuilder/pull/1869#discussion_r540080054